tutorials/tour/basics.md

```diff
@@ -0,0 +1,261 @@
+---
+layout: tutorial
+title: Basics
+
+disqus: true
+
+tutorial: scala-tour
+num: 2
+next-page: unified-types
+previous-page: tour-of-scala
+---
+
+In this page, we will cover basics of Scala.
+
+## Running Scala on Browser
+
+You can run Scala on browser by using [ScalaFiddle](https://scalafiddle.io).
+
+1. Go to https://scalafiddle.io.
+2. Copy and paste `println("Hello, world!")` to the left pane.
+3. Hit "Run" button and see it prints "Hello, world!" on the right pane.
+
+It is a perfect way for anybody to experiment a piece of Scala code anywhere, anytime!
+
+## Variables
+
+You can define variables with the `val` keyword.
+
+```
+val x = 1 + 1
+```
+
+Variables defined by `val` cannot be re-assigned and are immutable in that sense.
+
+```
+val x = 1 + 1
+x += 2 // This does not compile because it is re-assigning the variable.
+```
+
+If you need to mutate variables, you can use the `var` keyword instead.
+
+```
+var y = 1 + 1
+y += 2 // This compiles because "y" is declared with "var" keyword.
+```
+
+Type of variables can be inferred, but you can also explicitly state type like below.
+
+```
+val x: Int = 1 + 1
+```
+
+## Functions
+
+You can define functions that returns a given integer + 1 like below.
+
+```
+(x: Int) => x + 1
+```
+
+Left hand side of `=>` is parameter(s) and right hand side of it is body.
+
+Notice that the function has no `return` keyword. This is because, in Scala, the last expression of the function is automatically returned.
+
+You can also assign functions to variables.
+
+```
+val addOne = (x: Int) => x + 1
+println(addOne(1)) // 2
+```
+
+Or it can take multiple parameters.
+
+```
+val add = (x: Int, y: Int) => x + y
+println(add(1, 2)) // 3
+```
+
+Or it can take no parameters.
```

Comment:
Good point; it can be confusing.

I think methods taking multiple parameter groups is more advanced than function taking parameters. Maybe we should instead be explicit on methods section, explicitly stating difference between parameters and parameter groups. Then, readers can understand difference between "no parameters" vs. "no parameter groups" without confusing.